src/com/facebook/buck/jvm/java/BaseCompileToJarStepFactory.java

```diff
@@ -109,19 +116,51 @@ public void createCompileToJarStep(
    * is buck-out/bin/java/abc/lib__abc__classes/, then a contained class abc.AbcModule
    * should be at buck-out/bin/java/abc/lib__abc__classes/abc/AbcModule.class
    *
+   * @param context the build context.
+   * @param filesystem the project filesystem.
    * @param postprocessClassesCommands the list of commands to post-process .class files.
    * @param outputDirectory the directory that will contain all the javac output.
+   * @param declaredClasspathEntries the list of classpath entries.
    */
   @VisibleForTesting
   static ImmutableList<Step> addPostprocessClassesCommands(
-      Path workingDirectory,
+      BuildContext context,
+      ProjectFilesystem filesystem,
       List<String> postprocessClassesCommands,
-      Path outputDirectory) {
+      Path outputDirectory,
+      ImmutableSortedSet<Path> declaredClasspathEntries) {
     ImmutableList.Builder<Step> commands = new ImmutableList.Builder<Step>();
+
+    ImmutableMap.Builder<String, String> envVarBuilder = ImmutableMap.builder();
+    if (!postprocessClassesCommands.isEmpty()) {
+      envVarBuilder.put(
+          "CLASSPATH",
```

Comment:
Doesn't the "CLASSPATH" variable get special treatment by `java`?  Can this be renamed to something more specific like `COMPILATION_CLASSPATH`?